homeassistant/components/mercedesme.py

```diff
@@ -0,0 +1,149 @@
+
+    conf = config[DOMAIN]
+    username = conf.get(CONF_USERNAME)
+    password = conf.get(CONF_PASSWORD)
+    scan_interval = conf.get(CONF_SCAN_INTERVAL)
+
+    try:
+        mercedesme_api = Controller(username, password, scan_interval)
+        if not mercedesme_api.is_valid_session:
+            raise Exceptions.MercedesMeException(500)
+        hass.data[DATA_MME] = MercedesMeHub(mercedesme_api)
+    except Exceptions.MercedesMeException as ex:
+        if ex.code == 401:
+            hass.components.persistent_notification.create(
+                "Error:<br />Please check username and password."
+                "You will need to restart Home Assistant after fixing.",
+                title=NOTIFICATION_TITLE,
+                notification_id=NOTIFICATION_ID)
+        else:
+            hass.components.persistent_notification.create(
+                "Error:<br />Can't communicate with Mercedes me API.<br />"
+                "Error code: {} Reason: {}"
+                "You will need to restart Home Assistant after fixing."
+                "".format(ex.code, ex.message),
+                title=NOTIFICATION_TITLE,
+                notification_id=NOTIFICATION_ID)
+
+        _LOGGER.error("Unable to communicate with Mercedes me API: %s",
+                      ex.message)
+        return False
+
+    discovery.load_platform(hass, 'sensor', DOMAIN, {}, config)
+    discovery.load_platform(hass, 'device_tracker', DOMAIN, {}, config)
+    discovery.load_platform(hass, 'binary_sensor', DOMAIN, {}, config)
+
+    def hub_refresh(event_time):
+        """Call Mercedes me API to refresh information."""
+        _LOGGER.info("Updating Mercedes me component.")
+        hass.data[DATA_MME].data.update()
+        dispatcher_send(hass, SIGNAL_UPDATE_MERCEDESME)
+
+    track_time_interval(
+        hass,
+        hub_refresh,
+        timedelta(seconds=scan_interval))
+
+    return True
+
+
+class MercedesMeHub(object):
+    """Representation of a base MercedesMe device."""
+
+    def __init__(self, data):
+        """Initialize the entity."""
+        self.data = data
+
+
+class MercedesMeEntity(Entity):
+    """Entity class for MercedesMe devices."""
+
+    def __init__(self, data, internal_name, sensor_name, vin, unit):
+        """Initialize the MercedesMe entity."""
+        self._car = None
+        self._data = data
+        self._state = False
+        self._name = sensor_name
+        self._internal_name = internal_name
+        self._unit = unit
+        self._vin = vin
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @asyncio.coroutine
+    def async_added_to_hass(self):
+        """Register callbacks."""
+        async_dispatcher_connect(
+            self.hass, SIGNAL_UPDATE_MERCEDESME, self._update_callback)
+
+    def _update_callback(self):
+        """Callback update method."""
+        self.schedule_update_ha_state(True)
+
+    @property
+    def icon(self):

```

Comment:
This property can be removed.

Reply:
Once again, thank you for your review and the more than valid change requests. I've updated the code. Hope that we can land this in the next version. After that, I'll start to add new features.